### PR TITLE
lightsheet

### PR DESCRIPTION
**Pull request by mccalluc**
@cebriggs7135 : Lots of questions below that should be resolved before merge.

### CODE REVIEW

**Comment by mccalluc** [comment]
> Yes: https://drive.google.com/drive/folders/1qJ3_wq_qabT96AHAGWhBtFSMQCpPITgc?usp=sharing

- Is Alex aware of this, and will he be delivering a directory schema?
- The spec is not nearly specific enough to do anything with without additional conversation. While the TMCs were in the room it would have been good to pin to pin down *exactly* what the filenames are, and what kinds of variation are allowed. I expect that Alex will look at the output from one TMC, but from one example it's not clear what will vary between sites. In the future, please use [use this reference for how to describe directories](https://github.com/hubmapconsortium/ingest-validation-tools/blob/master/HOWTO-describe-directories.md).
- If at all possible, please do not supply textual information as an image. It makes it impossible for a downstream user to do anything with the information except by retyping it.
- If the assay description is complete, have you requested that it be imported to portal-docs? I believe we only want to reference portal-docs, not google docs.

File: src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - name: number_of_antibodies
+    description: Number of antibodies
+  - name: number_of_channels
+    description: Number of fluorescent channels imaged during each cycle.
+
+  # TODO: I'm not sure number_of_z_stacks belongs:
+  #   - Do you actually mean the number of layers in the stack, or are their multiple stacks?
+  #   - Why is the z dimension different from the x or y? We don't want to duplicate what can be inferred from the data.
+
+  # TODO: For consistency, can it just be "protocols_io_doi", without "acquisition_"?
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml:19]: I'm waiting for a response to your question from Marda.

**Review by cebriggs7135 — CHANGES_REQUESTED** [review]

**Comment by cebriggs7135** [comment]
@mccalluc  Melissa said on Jan 26th she'll have the Lightsheet description document in markdown soon.

File: src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+doc_url: https://portal.hubmapconsortium.org/docs/assays/lightsheet # TODO
+fields:
+  # TODO: Overrides of Level 1 fields
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml:3]: We always constraints on the values of assay_category, assay_type, and analyte_class. Without overriding the defaults, we get [every possible value](https://github.com/hubmapconsortium/ingest-validation-tools/pull/518/files#diff-a86bf9fa087edc0d62b8e4277a1db79ee2a7d56c36ce055f9d68f09919e2d897R121-R143).

File: src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  # include: ../includes/x_y_z.yaml
+
+  # TODO: Confirm descriptions / confirm that switching order for consistency is ok.
+  - name: number_of_antibodies
+    description: Number of antibodies
+  - name: number_of_channels
+    description: Number of fluorescent channels imaged during each cycle.
+
+  # TODO: I'm not sure number_of_z_stacks belongs:
+  #   - Do you actually mean the number of layers in the stack, or are their multiple stacks?
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml:16]: Right, but that is a characteristic of the data that can be found by reading the data. Specifying it additionally as metadata makes inconsistencies likely.

File: src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml (hunk 1 above)
Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml:19]: For changes to the text, please use the github machinery to suggest edits on the yaml file.

File: src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml:3]: @mccalluc 
Lightcsheet assay_category is "imaging".
Assay_type is "Light Sheet".
Analyte_class is "protein".

**Comment by mccalluc** [comment]
@cebriggs7135 : Can you review this, in particular, the [generated docs](https://github.com/hubmapconsortium/ingest-validation-tools/blob/25d32dce6b5e0f47f83d4556e6949098ed89f15e/docs/lightsheet/README.md). 
- Check that the documentation around z layers seems right to you.
- There is currently only one protocols_io reference... but perhaps there should be a separate one for "aquisition"?

File: src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml (hunk 1 above)
Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/level-2/lightsheet.yaml]: I'll work on this today.

**Comment by cebriggs7135** [comment]
@mccalluc Re: the description of z value, the document says distance between slices but that's not correct so I'll fix that today. 
Re: the acquisition protocol, UFL (Seth & Marda) say that field is not necessary.

**Pull request merged**